src/service/data_products/heatmap.py

```diff
@@ -317,54 +319,45 @@ async def _get_load_version_and_processes( # pretty huge method sig here
         self,
         appstate: CollectionsState,
         user: kb_auth.KBaseUser | None,
+        collection: str,
         collection_id: str,
-        data_product: str,  # can use self._id here but if we want to make this general in future
-        subset_fn: Callable[
-            [
-                PickleableDependencies,
-                ArangoStorage,
-                models.InternalMatch | models.InternalSelection,
-                models.SavedCollection,
-                models.DataProductProcessIdentifier
-            ],
-            Awaitable[None],
-        ],
+        data_product: str,
         load_ver_override: str | None = None,
         match_id: str | None = None,
         selection_id: str | None = None,
     ):
         # this is very similar to code in taxa_counts - maybe once it gets cleaned up a bit
         # and handles the dependency on genome_attribs in a saner way this can be moved
-        # to common_functions and shared
+        # to common_functions and shared. Would need to be able to specify its own subsetting fn
         dp_match, dp_sel = None, None
-        if match_id or selection_id:
-            coll = await appstate.arangostorage.get_collection_active(collection_id)
-            load_ver = get_load_ver_from_collection(coll, data_product)
-        else:
-            load_ver = await get_load_version(
-                appstate.arangostorage, collection_id, data_product, load_ver_override, user)
+        lvo = override_load_version(load_ver_override, match_id, selection_id)
+        coll, load_ver = await get_load_version(
+            appstate.arangostorage, collection_id, data_product, lvo, user)
         if match_id:
             dp_match = await processing_matches.get_or_create_data_product_match_process(
-                appstate, coll, user, match_id, data_product, subset_fn
+                appstate, coll, user, match_id, data_product,
+                partial(self._process_subset, collection)
```

Comment:
I'll get rid of `self` references in this method when I move it and `_process_subset` to a new module

Reply:
👍